### PR TITLE
Backend/task ownership

### PR DESCRIPTION
**Pull request by CalvinDong**

### CODE REVIEW

File: backend/src/routes/task-router.ts [hunk 1]
```diff
@@ -35,7 +35,15 @@ export const TaskRoute = (app: Express, controller: TaskController) => {
      *         example: Boil water
      *        recipe:
      *         type: string
-     *         example: recipe01
+     *         example: {
```
Comment (mitcmurph) [inline backend/src/routes/task-router.ts:38]: Did you change this because the old line did not work? I think I broke it, Tong will fix it.

**Review by mitcmurph — CHANGES_REQUESTED** [review]
Can you please add one more thing. When creating a task set the new field to the id of the user making the request.

File: backend/src/routes/task-router.ts (hunk 1 above)
Comment (CalvinDong) [reply backend/src/routes/task-router.ts:38]: yeah lol. Do you want to change it back to how it was before?

**Review by CalvinDong — COMMENTED** [review]
> Can you please add one more thing. When creating a task set the new field to the id of the user making the request.

Do you mean adding one in for the example? Users should already be able to add the their id to the task

**Comment by CalvinDong** [comment]
- the just commit now push is just some typescript stuff

File: backend/src/controllers/task.ts [hunk 2]
```diff
@@ -34,11 +59,19 @@ export default class TaskController {
   // Create a task
   public async create(req: Request, res: Response) {
     const { body } = req;
+    let user: any
+    if (req.user){
```
Comment (mitcmurph) [inline backend/src/controllers/task.ts:63]: In future can use optional chaining as a shortcut for this. `req?.user._id`

**Review by mitcmurph — APPROVED** [review]

**Pull request merged**